builtin/credential/aws/path_role.go

```diff
@@ -892,9 +892,11 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
 	defaultLeaseTTL := b.System().DefaultLeaseTTL()
 	systemMaxTTL := b.System().MaxLeaseTTL()
 	if roleEntry.TokenTTL > defaultLeaseTTL {
+		roleEntry.TokenTTL = defaultLeaseTTL
 		resp.AddWarning(fmt.Sprintf("Given ttl of %d seconds greater than current mount/system default of %d seconds; ttl will be capped at login time", roleEntry.TokenTTL/time.Second, defaultLeaseTTL/time.Second))
 	}
 	if roleEntry.TokenMaxTTL > systemMaxTTL {
+		roleEntry.TokenMaxTTL = systemMaxTTL
 		resp.AddWarning(fmt.Sprintf("Given max ttl of %d seconds greater than current mount/system default of %d seconds; max ttl will be capped at login time", roleEntry.TokenMaxTTL/time.Second, systemMaxTTL/time.Second))
```

Comment:
> max ttl will be capped at login time

Is this now slightly inaccurate? I believe if you read the config back for the role, it will probably have stored the system max TTL instead of the specified max TTL right? So it's actually already been capped.

Reply:
Right, we should remove the about "at login". We're now setting this on the role.

Reply:
> I believe if you read the config back for the role, it will probably have stored the system max TTL instead of the specified max TTL right?

I don't think that this is the case because we're not actually modifying/overriding `roleEntry.TokenMaxTTL` so the cap is actually determined at login time. The outdated diff looks like we were at one point but then we reverted back (which I think is the right call).